### PR TITLE
allow custom header on pause, resume recording

### PR DESCRIPTION
**Pull request by xquanluu**
waiting them to test on real env

### CODE REVIEW

**Comment by xquanluu** [comment]
They confirmed feature is working for re-invite custom headers on recording command

File: lib/session/call-session.js [hunk 1]
```diff
@@ -622,8 +622,11 @@ class CallSession extends Emitter {
       const res = await this.dlg.request({
         method: 'INFO',
         headers: {
-          'X-Reason': 'pauseCallRecording'
-        }
+          'X-Reason': 'pauseCallRecording',
+          ...(this.recordOptions.headers && {'Content-Type': 'application/json'})
```
Comment (davehorton) [inline lib/session/call-session.js:626]: are we guaranteed that this.recordOptions will exist?

File: lib/session/call-session.js [hunk 2]
```diff
@@ -622,8 +622,11 @@ class CallSession extends Emitter {
...
+        },
+        // Siprect Client is initiated from startCallRecording, so just need to pass custom headers in startRecording
```
Comment (davehorton) [inline lib/session/call-session.js:628]: minor typo "Siprect"

File: lib/session/call-session.js (hunk 1 above)
Comment (xquanluu) [reply lib/session/call-session.js:626]: https://github.com/jambonz/jambonz-feature-server/blob/main/lib/session/call-session.js#L549

before executing recording command, recordOptions = data from recording command. so it always available for recording command

**Review by davehorton — APPROVED** [review]

**Pull request merged**